### DISCUSSION TITLE
4.0.0 - package namespace rename

### DISCUSSION THREAD

**Discussion by Stormheg** [django-recaptcha/django-recaptcha#319] [Announcements]
## Package namespace renamed!

The package namespace `captcha` has been renamed to `django_recaptcha` to avoid conflicts with other captcha packages.

[Please see the full CHANGELOG for all changes and upgrade considerations](https://github.com/torchbox/django-recaptcha/blob/62afff4c109cf9b918b4a91e7db5981ff4757842/CHANGELOG.md#400-2023-11-14)

## What's Changed
* .pre-commit-config.yaml: Remove duplicate checks by @cclauss in https://github.com/torchbox/django-recaptcha/pull/279
* Fix typos discovered by codespell by @cclauss in https://github.com/torchbox/django-recaptcha/pull/278
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/torchbox/django-recaptcha/pull/273
* moved recaptcha execute into event listener function on form submit by @thbarrons in https://github.com/torchbox/django-recaptcha/pull/296
* Add testing with Django 4.1 by @matthiask in https://github.com/torchbox/django-recaptcha/pull/293
* Added example unittest for RECAPTCHA_TESTING by @guettli in https://github.com/torchbox/django-recaptcha/pull/289
* Update testing instructions by @jnns in https://github.com/torchbox/django-recaptcha/pull/300
* Add Indonesian (id) translations by @kiraware in https://github.com/torchbox/django-recaptcha/pull/301
* Add Django 4.2 and Python 3.11 testing by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/305
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/torchbox/django-recaptcha/pull/286
* Use CodeQL v2 action by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/306
* Update trove classifiers by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/307
* Feature Request: Add actions in recaptcha v3 widget by @alexrobertson-muckrack in https://github.com/torchbox/django-recaptcha/pull/304
* Documentation for V3 action parameter by @alexrobertson-muckrack in https://github.com/torchbox/django-recaptcha/pull/309
* Move class to widget by @rgs258 in https://github.com/torchbox/django-recaptcha/pull/275
* Fix #280: Mark ReCaptchaV3 as a hidden field by @matthiask in https://github.com/torchbox/django-recaptcha/pull/294
* V3 action Verification by @alexrobertson-muckrack in https://github.com/torchbox/django-recaptcha/pull/310
* Added Arabic translations by @YDA93 in https://github.com/torchbox/django-recaptcha/pull/313
* Add ukrainian translation by @koutoftimer in https://github.com/torchbox/django-recaptcha/pull/315
* Fix invalid RST link by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/311
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/torchbox/django-recaptcha/pull/308
* Mark wheel as not compatible with python2 by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/272
* Rename package to `django_recaptcha` by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/317
* Prepare v4 release by @Stormheg in https://github.com/torchbox/django-recaptcha/pull/312

## New Contributors
* @cclauss made their first contribution in https://github.com/torchbox/django-recaptcha/pull/279
* @thbarrons made their first contribution in https://github.com/torchbox/django-recaptcha/pull/296
* @matthiask made their first contribution in https://github.com/torchbox/django-recaptcha/pull/293
* @guettli made their first contribution in https://github.com/torchbox/django-recaptcha/pull/289
* @jnns made their first contribution in https://github.com/torchbox/django-recaptcha/pull/300
* @kiraware made their first contribution in https://github.com/torchbox/django-recaptcha/pull/301
* @alexrobertson-muckrack made their first contribution in https://github.com/torchbox/django-recaptcha/pull/304
* @rgs258 made their first contribution in https://github.com/torchbox/django-recaptcha/pull/275
* @YDA93 made their first contribution in https://github.com/torchbox/django-recaptcha/pull/313
* @koutoftimer made their first contribution in https://github.com/torchbox/django-recaptcha/pull/315

**Full Changelog**: https://github.com/torchbox/django-recaptcha/compare/3.0.0...4.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/torchbox/django-recaptcha/releases/tag/4.0.0'>4.0.0 - package namespace rename</a>.</em>

**Comment by Stormheg** [comment]
This is the first time we release this package to PyPI using GitHub Actions and something is going wrong: the publish workflow is not triggering! 🤯 

Looking at the repository settings, it also appears we are missing a token to publish to PyPI. Will get in touch with Torchbox to get this sorted as they literally hold all the keys to PyPI 🔑

This will delay the release to PyPI a bit... will try to get it sorted out asap!

**Reply by Stormheg** [reply]
It's out people! https://pypi.org/project/django-recaptcha/4.0.0/

**Reply by enzedonline** [reply]
Thank you! Working great. :) 

Heads up for anyone using wagtailcaptcha, import on line 4, wagtailcaptcha/forms.py needs updating to the new namespace.

**Reply by cclauss** [reply]
No Python 3.12 support?

**Reply by koutoftimer** [reply]
@cclauss I would expect support of the optional GIL disabling.

**Reply by Stormheg** [reply]
@cclauss Python 3.12 support is not official because the latest supported Django is 4.2, which has no official Python 3.12 support so this package doesn't either.

With Django 5.0 scheduled to be released next month, official support for Python 3.12 will be added too 👍

**Comment by cclauss** [comment]
Yes.  https://code.djangoproject.com/ticket/34118 was an over-promise

**Reply by Stormheg** [reply]
Once Django 4.2 + Python 3.12 support is official we will start supporting it officially too.

**Comment by thibaudcolas** [comment]
With 4.0.0 out of the way – we’re looking for more maintainers to reduce the bus factor on this package. If anyone here is interested, please reply on [New maintainers – and more needed #249](https://github.com/torchbox/django-recaptcha/discussions/249#discussioncomment-7617875).
